applications/GeoMechanicsApplication/custom_constitutive/small_strain_umat_law.hpp

```diff
@@ -118,22 +123,21 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) SmallStrainUMAT3DLaw : public Consti
     /// Static definition of the dimension
     static constexpr SizeType Dimension = N_DIM_3D;
 
-    /// Static definition of the VoigtSize
-    static constexpr SizeType VoigtSize = VOIGT_SIZE_3D;
-
-    /// Pointer definition of SmallStrainUMAT3DLaw
-    KRATOS_CLASS_POINTER_DEFINITION(SmallStrainUMAT3DLaw);
+    /// Pointer definition of SmallStrainUMATLaw
+    KRATOS_CLASS_POINTER_DEFINITION(SmallStrainUMATLaw);
 
     //@}
     //@name Life Cycle
     //@{
 
-    SmallStrainUMAT3DLaw()           = default;
-    ~SmallStrainUMAT3DLaw() override = default;
-    SmallStrainUMAT3DLaw(const SmallStrainUMAT3DLaw& rOther);
-    SmallStrainUMAT3DLaw& operator=(const SmallStrainUMAT3DLaw& rOther);
-    SmallStrainUMAT3DLaw(SmallStrainUMAT3DLaw&&)            = delete;
-    SmallStrainUMAT3DLaw& operator=(SmallStrainUMAT3DLaw&&) = delete;
+    SmallStrainUMATLaw() = default;
+    explicit SmallStrainUMATLaw(std::unique_ptr<ConstitutiveLawDimension> pConstitutiveDimension);
+
+    ~SmallStrainUMATLaw() override;
+    SmallStrainUMATLaw(const SmallStrainUMATLaw& rOther);
+    SmallStrainUMATLaw& operator=(const SmallStrainUMATLaw& rOther);
+    SmallStrainUMATLaw(SmallStrainUMATLaw&&)            = delete;
+    SmallStrainUMATLaw& operator=(SmallStrainUMATLaw&&) = delete;
```

Comment:
As of C++17, [move operations should be marked `noexcept`](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c66-make-move-operations-noexcept):
```suggestion
    SmallStrainUMATLaw(SmallStrainUMATLaw&&) noexcept            = delete;
    SmallStrainUMATLaw& operator=(SmallStrainUMATLaw&&) noexcept = delete;
```

Reply:
Done